jme3-core/src/main/java/com/jme3/scene/shape/CenterQuad.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.scene.shape;
+
+import com.jme3.export.InputCapsule;
+import com.jme3.export.JmeExporter;
+import com.jme3.export.JmeImporter;
+import com.jme3.export.OutputCapsule;
+import com.jme3.scene.Mesh;
+import com.jme3.scene.VertexBuffer.Type;
+import java.io.IOException;
+
+/**
+ * A static, indexed, Triangles-mode mesh for an axis-aligned rectangle in the
+ * X-Y plane.
+ *
+ * <p>The rectangle extends from (-width/2, -height/2, 0) to
+ * (width/2, height/2, 0) with normals set to (0,0,1).
+ *
+ * <p>This differs from {@link com.jme3.scene.shape.Quad} because it puts
+ * (0,0,0) at the rectangle's center instead of in a corner.
+ *
+ * @author Kirill Vainer
+ */
+public class CenterQuad extends Mesh {
+
+    private float width;
+    private float height;
+
+    /**
+     * For serialization only. Do not use.
+     */
+    protected CenterQuad() {
+    }
+
+    /**
+     * Instantiate an unflipped quad in the X-Y plane with the specified width
+     * and height.
+     *
+     * @param width the desired X extent or width
+     * @param height the desired Y extent or height
+     */
+    public CenterQuad(float width, float height) {
+        this(width, height, false);
+    }
+
+    /**
+     * Instantiate a quad in the X-Y plane with the specified width and height.
+     *
+     * @param width the desired X extent or width
+     * @param height the desired Y extent or height
+     * @param flipCoords true to flip the texture coordinates (v=0 when
+     *     y=height/2) or false to leave them unflipped (v=1 when y=height/2)
+     */
+    public CenterQuad(float width, float height, boolean flipCoords) {
+        updateGeometry(width, height, flipCoords);
+        super.setStatic();
```

Comment:
This instruction can be removed. The `setsStatic()` method is already invoked in the `updateGeometry()` method.

Reply:
Thanks for the tip. I'll do that.